libyul/YulString.h

```diff
@@ -79,8 +80,29 @@ class YulStringRepository: boost::noncopyable
 		return hash;
 	}
 	static constexpr std::uint64_t emptyHash() { return 14695981039346656037u; }
+	/// Clear the repository.
+	/// Use with care - there cannot be any dangling YulString references.
+	/// If references need to be cleared manually, register the callback via
+	/// resetCallback.
+	static void reset()
+	{
+		for (auto const& cb: resetCallbacks())
+			cb();
+		instance() = YulStringRepository{};
+	}
+	static std::vector<std::function<void()>>& resetCallbacks()
+	{
+		static std::vector<std::function<void()>> callbacks;
```

Comment:
How are `callbacks` reset? Can't they accumulate over time?

Reply:
I think they are just registered once, but I can check again.

Reply:
Since the fuzzer calls this function twice every iteration:

https://github.com/ethereum/solidity/blob/6e9bb2c07304deba09a36b31cde49fa5895d66c2/test/tools/ossfuzz/yulProto_diff_ossfuzz.cpp#L82

Could memory get accumulated here?

https://github.com/ethereum/solidity/blob/7d9007dc04554710cc8b3ff7e6370503728ad332/libyul/backends/evm/EVMDialect.cpp#L203

Reply:
Good catch, this is an issue!